packages/boot/test/bootstrapTests/supports.js

```diff
@@ -426,7 +426,7 @@ export const makeSwingsetTestKit = async (
     bridgeOutbound,
     kernelStorage,
     configPath,
-    {},
+    [],
```

Comment:
I suspect that type spec is wrong.  It's an array.

Reply:
I learned the hard way, 6 months ago (ce4aa3cb29b0b), that it's not an array:

If we trace back the callers, we find `argv` is bound to `action` in...

https://github.com/Agoric/agoric-sdk/blob/8fc0c23551a496de4b55068657cc94b4ddec1839/packages/cosmic-swingset/src/chain-main.js#L677

and just above that, we see that `action` is not an array:

https://github.com/Agoric/agoric-sdk/blob/8fc0c23551a496de4b55068657cc94b4ddec1839/packages/cosmic-swingset/src/chain-main.js#L644-L650

It comes from...

https://github.com/Agoric/agoric-sdk/blob/8fc0c23551a496de4b55068657cc94b4ddec1839/golang/cosmos/app/app.go#L983-L996

Reply:
Oh dear.  We might have a problem (well, probably we don't because the chain _is_ running) but various places in the kernel really expect an array there.

Reply:
Glad we've got these type annotations to reveal the discrepancy!

Reply:
OK, I think I see what's going on here: confusion between vat parameters in general and the specific `argv` (often loosely referred to as "args") vat parameter.  I will investigate further and determine if the confusion is mine or somebody else's.  Flip of a coin at this point, though I'm hoping the confusion is mine because it's the easiest to fix.

Reply:
Digging through the kernel implementation, I see that it's an array of strings, and in fact is declared as such in a number of places.  However, it's clear that those type definitions are not wired through to the places where it was declared as `unknown` and so the `unknown` did not get flagged by `tsc` as a conflicting type definition.  The question that I'm not sure I know the answer to, since I don't completely follow @dckc's forensics there, is whether we actually have places that use it in the form of something other than an array of strings. This is the kind of thing one can often get away with given JavaScript's loosey-goosey approach to runtime typing in which as long the producer of the datum and the user of the datum have a shared belief about what it is, everything comes out OK regardless of what anybody else along the datapath from point A to point B believes.

Reply:
> whether we actually have places that use it in the form of something other than an array of strings

Yes.  Since the earliest days of SwingSet, we've had host programs that used an arbitrary jsonable object, not (necessarily) an array.   See [`ag-chain-cosmos`'s `argv`](https://github.com/Agoric/agoric-sdk/blob/46a7b2e0947d2110bdee094dee0cf72aa7a13816/packages/cosmic-swingset/src/chain-main.js#L359),  [`ag-solo`'s `argv`](https://github.com/Agoric/agoric-sdk/blob/46a7b2e0947d2110bdee094dee0cf72aa7a13816/packages/solo/src/main.js#L131), and [the sim chain's `argv`](https://github.com/Agoric/agoric-sdk/blob/46a7b2e0947d2110bdee094dee0cf72aa7a13816/packages/cosmic-swingset/src/sim-chain.js#L64).

Reply:
I don't feel strongly enough to request changes.
I have already deferred to someone else to approve the whole PR. I'll leave this to others too.

Reply:
Let's make some change to avoid further confusion. I think we agree that's it not worth the churn to change "argv" to be an array.

I hope we can agree that "argv", short for "argument vector", is a confusing name for non-arrays.

If so, I propose that we make the type 'unknown' consistently and take the "v" out of the name. It could be in a different PR. Meanwhile this one shouldn't change the {} because it is valid.

Reply:
> Let's make some change to avoid further confusion.

Changing the name `argv` looks like a huge amount of work, to me. I think it involves upgrading the bootstrap vat, which is sort of an open problem.

This is not part of the API surface that contract devs etc. are likely to run into. The cost of the confusion is moderate. I strongly suggest we just accept it.

Reply:
OK, I'm going to treat this conversation as resolved and leave the code as is.